### ISSUE TITLE
Pyramid image fails for non strictly decreasing tiles.

### ISSUE THREAD

**Issue by manzt** [napari/napari#984]
## 🐛 Bug

Trying to build a zarr pyramid and lazily load with dask. I padded some of the smallest tiles to be 512 x 512, but since layers are not "strictly decreasing" in shape this errors despite `is_pyramid` passed explicitly.

## To Reproduce

1. Create a (mostly) decreasing pyramid, and explicitly pass `is_pyramid` to `viewer.add_image`.
```python
from pathlib import Path
import dask.array as da
import napari

path = Path.cwd() / "my_dataset.zarr" / "mxif_pyramid"
max_level = 11
pyramid = [da.from_zarr(pyramid_path, component=str(i).zfill(2)) for i in range(max_level)]
print([layer.shape for layer in pyramid])
# [(36040, 52660),
# (18020, 26330),
# (9010, 13165),
# (4505, 6582),
# (2252, 3291),
# (1126, 1645),
# (563, 822),
# (512, 512),
# (512, 512),
# (512, 512),
# (512, 512)]
viewer = napari.Viewer()
viewer.add_image(pyramid, is_pyramid=True)
```

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-9-7d5ed5eb6088> in <module>
----> 1 viewer.add_image(pyramid, is_pyramid=True)

~/dev/miniconda3/envs/py3.8/lib/python3.8/site-packages/napari/components/viewer_model.py in add_image(self, data, channel_axis, rgb, is_pyramid, colormap, contrast_limits, gamma, interpolation, rendering, iso_threshold, attenuation, name, metadata, scale, translate, opacity, blending, visible, path)
    500             if blending is None:
    501                 blending = 'translucent'
--> 502             layer = layers.Image(
    503                 data,
    504                 rgb=rgb,

~/dev/miniconda3/envs/py3.8/lib/python3.8/site-packages/napari/layers/image/image.py in __init__(self, data, rgb, is_pyramid, colormap, contrast_limits, gamma, interpolation, rendering, iso_threshold, attenuation, name, metadata, scale, translate, opacity, blending, visible)
    160             data = list(data)
    161
--> 162         ndim, rgb, is_pyramid, data_pyramid = get_pyramid_and_rgb(
    163             data, pyramid=is_pyramid, rgb=rgb
    164         )

~/dev/miniconda3/envs/py3.8/lib/python3.8/site-packages/napari/layers/image/image_utils.py in get_pyramid_and_rgb(data, pyramid, rgb)
    132         init_shape = shapes[0]
    133     else:
--> 134         init_shape = data.shape
    135
    136     # Determine if rgb, and determine dimensionality

AttributeError: 'list' object has no attribute 'shape'
```

But this works as expected

```python
from pathlib import Path
import dask.array as da
import napari

path = Path.cwd() / "my_dataset.zarr" / "mxif_pyramid"
max_level = 7
pyramid = [da.from_zarr(pyramid_path, component=str(i).zfill(2)) for i in range(max_level)]
print([layer.shape for layer in pyramid])
# [(36040, 52660),
# (18020, 26330),
# (9010, 13165),
# (4505, 6582),
# (2252, 3291),
# (1126, 1645),
# (563, 822),
# (512, 512)]
viewer = napari.Viewer()
viewer.add_image(pyramid, is_pyramid=True)
```

## Expected behavior

Open the image stack since `is_pyramid` is passed explicitly.

## Environment
* `napari 0.2.10`
* `python 3.8`

**Comment by manzt** [comment]
Lesson learned that the final tiles aren't necessary, but the breaking behavior is unexpected. Perhaps a more informative error message that tiles must be strictly decreasing would be helpful. It took a little while to debug.

**Comment by sofroniewn** [comment]
I think a more informative error message is the way to go here, as I'm not sure why you would need the final tiles - but let me know @manzt if you have a need for them.

We could also put a warning in here https://github.com/napari/napari/blob/master/napari/layers/image/image_utils.py#L17 if we think that something is close to a pyrmaid, but not quite.

**Comment by manzt** [comment]
I definitely don't think they are necessary to have, and padding images is a pain anyways. No need for the final tiles!

> We could also put a warning in here https://github.com/napari/napari/blob/master/napari/layers/image/image_utils.py#L17 if we think that something is close to a pyrmaid, but not quite.

A warning would have been great. A colleague told me to check the shape of each array and helped me to debug. The `AttributeError: 'list' object has no attribute 'shape'` made me think there was an issue with passing a list to `viewer.add_image` rather than an array directly. Might just be me but making that more clear would be awesome!

**Comment by sofroniewn** [comment]
Yup, ok that makes sense - we'll get something in. Thanks for letting us know

**Comment by manzt** [comment]
Thank you!

**Comment by sofroniewn** [comment]
non strictly decreasing tiles are still not supported, but this can now be closed by #1244 which adds more much informative error message.
